nemo/collections/asr/parts/submodules/conformer_modules.py

```diff
@@ -105,15 +137,25 @@ def forward(self, x, att_mask=None, pos_emb=None, pad_mask=None):
 
         x = self.norm_self_att(residual)
         if self.self_attention_model == 'rel_pos':
-            x = self.self_attn(query=x, key=x, value=x, mask=att_mask, pos_emb=pos_emb)
+            x = self.self_attn(
+                query=x,
```

Comment:
Maybe worth it to leave original MHA class alone, subclass it and add such caching support. Otherwise it will just get more complicated over time.

Reply:
These new inputs don't change the behavior of the current modules when not passed. It just add a new update_cache method and call at the beginning. Adding a new class for such an addition looks too much.